jina/helper.py

```diff
@@ -693,12 +697,16 @@ def run(self):
         return asyncio.run(func(*args, **kwargs))
 
 
-def check_keys_exist(keys_to_check, existing_keys):
-    missed = []
-    for k in keys_to_check:
-        if k not in existing_keys:
-            missed.append(k)
-    return missed
+def filter_nonexistent_keys(keys: Iterator, existent_keys: Iterator, check_path: str, logger: 'JinaLogger'):
```

Comment:
I would prefer having this method as a part of a base class of all the using classes. Might be `BaseIndexer`? Having the logger as an argument is a smell in my opinion.

Reply:
I agree

Reply:
Ah good point. Done